shelly/src/main/java/au/com/jtribe/shelly/Alarm.java

```diff
@@ -0,0 +1,219 @@
+    checkNotNull(message, "message == null");
+    this.message = message;
+    return this;
+  }
+
+  /**
+   * Adds the hour that the alarm should be set to. Should be a value between 0-23
+   *
+   * @param hour hour of the day to set the alarm to
+   * @return Object this method was called on for method chaining.
+   */
+  @NonNull
+  @CheckResult
+  public Alarm hour(@NonNull Integer hour) {
+    checkNotNull(hour, "hour == null");
+    //if ( hour > 24 || hour < 0 )  // TODO: validate the hour
+    this.hour = hour;
+    return this;
+  }
+
+  /**
+   * Adds the minute that the alarm should be set to. Should be a value between 0-59
+   *
+   * @param minute minute of the day to set the alarm to
+   * @return Object this method was called on for method chaining.
+   */
+  @NonNull
+  @CheckResult
+  public Alarm minute(@NonNull Integer minute) {
+    checkNotNull(minute, "minute == null");
+    //if ( minute > 60 || minute < 0 )  // TODO: validate the hour
+    this.minute = minute;
+    return this;
+  }
+
+  /**
+   * SkipUI allows an alarm to be set without opening the Alarm activity.
+   * If this is not set or is set to false the Alarm activity will be opened
+   *
+   * @param skipUi when true the alarm Activity won't be opened.
+   * @return Object this method was called on for method chaining.
+   */
+  @NonNull
+  @CheckResult
+  public Alarm skipUi(@NonNull Boolean skipUi) {
+    checkNotNull(skipUi, "skipUi == null");
+    this.skipUi = skipUi;
+    return this;
+  }
+
+  /**
+   * Set custom ringTone for the alarm
+   * See MainActivity examples
+   *
+   * @param ringTone URI to the ringTone to play when the alarm goes off
+   * @return Object this method was called on for method chaining.
+   */
+  @NonNull
+  @CheckResult
+  public Alarm ringTone(@NonNull Uri ringTone) {
+    checkNotNull(ringTone, "ringTone == null");
+    this.ringTone = ringTone;
+    return this;
+  }
+
+  /**
+   * Option to silence the ringTone turning the alarm into a vibrate only alarm
+   *
+   * @return Object this method was called on for method chaining.
+   */
+  @NonNull
+  @CheckResult
+  public Alarm silenceRingTone(@NonNull Boolean silenceRingTone) {
+    checkNotNull(silenceRingTone, "silenceRingTone == null");
+    this.silenceRingTone = silenceRingTone;
+    return this;
+  }
+
+  /**
+   * Option to enable vibrate on the alarms, by default alarms vibrate
+   *
+   * @return Object this method was called on for method chaining.
+   */
+  @NonNull
+  @CheckResult
+  public Alarm vibrate(@NonNull Boolean vibrate) {
+    checkNotNull(vibrate, "vibrate == null");
+    this.vibrate = vibrate;
+    return this;
+  }
+
+  /**
+   * Used to created a repeating alarm, repeat on these days of the week.
+   *
+   * @param days Accepts days as Integers, use Calendar.MONDAY to do this
+   * @return Object this method was called on for method chaining.
+   */
+  @NonNull
+  @CheckResult
+  public Alarm daysToRepeat(@NonNull Integer... days) {
```

Comment:
if you change this to accept varargs of int instead you don't need to do the null check. Maybe we should have a discussion about whether we want to take primitives instead of boxed values whenever possible? What do you think, @AngusMorton?